openai/openai.go

```diff
@@ -269,10 +271,52 @@ func toModel(r api.ShowResponse, m string) Model {
 	}
 }
 
-func fromChatRequest(r ChatCompletionRequest) api.ChatRequest {
+func fromChatRequest(r ChatCompletionRequest) (api.ChatRequest, error) {
 	var messages []api.Message
 	for _, msg := range r.Messages {
-		messages = append(messages, api.Message{Role: msg.Role, Content: msg.Content})
+		switch content := msg.Content.(type) {
+		case string:
+			messages = append(messages, api.Message{Role: msg.Role, Content: content})
+		case []any:
+			message := api.Message{Role: msg.Role}
+			for _, c := range content {
+				data, ok := c.(map[string]any)
+				if !ok {
+					return api.ChatRequest{}, fmt.Errorf("invalid message format")
+				}
+				switch data["type"] {
+				case "text":
+					text, ok := data["text"].(string)
+					if !ok {
+						return api.ChatRequest{}, fmt.Errorf("invalid message format")
+					}
+					message.Content = text
+				case "image_url":
```

Comment:
We can merge this as is, however with @mxyng 's new messages changes that collate messages, it would be better to instead create individual messages for each text and image content type we see

Reply:
will keep in mind for a potential refactor